### ISSUE TITLE
Run Slice fails

### ISSUE THREAD

**Issue by kujaku11** [kujaku11/mth5#85]
When asking for a slice within a run get an error.

@kkappler can you add in the error and how you got to the error.

**Comment by kkappler** [comment]
Opening an mth5 and taking the channel summary:

```
# mth5_path = "data/ZU_CAS04.h5"
from mth5.utils.helpers import initialize_mth5
mth5_obj = initialize_mth5(mth5_path, "r")
ch_sum = mth5_obj.channel_summary
run_a = mth5_obj.get_run("CAS04", "a")
run_ts_01 = run_a.to_runts()
run_ts_02 = run_a.to_runts(ch_sum.iloc[0].start, end=ch_sum.iloc[0].end)
```
run_ts_01 was fine, but run_ts_02 encountered a RuntimeError:

```
File "/home/kkappler/software/irismt/mth5/mth5/groups/master_station_run_channel.py", line 1361, in to_runts
    ts_obj = ch_obj.time_slice(start, end=end, n_samples=n_samples)
  File "/home/kkappler/software/irismt/mth5/mth5/groups/master_station_run_channel.py", line 2549, in time_slice
    regional_ref = self.hdf5_dataset.regionref[start_index : end_index + 1]
  File "/home/kkappler/anaconda2/envs/py37/lib/python3.7/site-packages/h5py/_hl/base.py", line 257, in __getitem__
    return h5r.create(self.id, b'.', h5r.DATASET_REGION, selection.id)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5r.pyx", line 63, in h5py.h5r.create
RuntimeError: Unable to encode region reference (no write intent on file)
```

**Comment by kkappler** [comment]
To reproduce the error you can paste the following into 
aurora/tests/synthetic/ as say, run_ts_test_slice.py

```
from mth5.utils.helpers import initialize_mth5

mth5_path = "data/test1.h5"
mth5_obj = initialize_mth5(mth5_path, "r")
ch_sum = mth5_obj.channel_summary
run_001 = mth5_obj.get_run("test1", "001")
run_ts_01 = run_001.to_runts()
run_ts_02 = run_001.to_runts(ch_sum.iloc[0].start, end=ch_sum.iloc[0].end)
```

**Comment by kujaku11** [comment]
@kkappler The issue here is that the file is opened in mode `r`.  The way I had written slice was to create an internal hdf5 reference of the requested slice and since the file was in read mode it couldn't create that reference.  I added a try/except for read mode and if it is in read mode then a simple `slice()` is used to get the requested data.  This is fixed in PR #87
